components/infobox/wikis/trackmania/infobox_league_custom.lua

```diff
@@ -140,10 +146,38 @@ function CustomLeague:addToLpdb(lpdbData, args)
 
 	lpdbData.maps = table.concat(self:getAllArgsForBase(args, 'map'), ';')
 
+	lpdbData.extradata.circuit = args.circuit
+	lpdbData.extradata.circuittier = args.circuittier
+
 	-- Legacy, can be superseeded by lpdbData.mode
 	lpdbData.extradata.individual = self.data.mode == DEFAULT_MODE
 
 	return lpdbData
 end
 
+---@param widgets Widget[]
+function CustomLeague:_createCircuitInformation(widgets)
+	local args = self.args
+
+	Array.appendWith(widgets,
+		Cell{
+			name = 'Circuit',
+			content = {self:_createCircuitLink()}
+		},
+		Cell{name = 'Circuit Tier', content = {args.circuittier}},
+		Cell{name = 'Tournament Region', content = {args.region}},
+		Chronology{content = {next = args.circuit_next, previous = args.circuit_previous}}
+	)
+end
+
+---@return string?
+function CustomLeague:_createCircuitLink()
+	local args = self.args
+
+	return self:createSeriesDisplay({
+		displayManualIcons = true,
+		series = args.circuit,
+	}, self.data.circuitIconDisplay)
```

Comment:
Where is the circuitIconDisplay coming from?

Reply:
I don't think it's really come from anywhere. It's the storage for the circuitIconDisplay param from the createSeriesDisplay call. It should work even without the circuitIconDisplay call

Reply:
Then i'd  say remove it
```suggestion
	})
```